resources/views/settings/slack.blade.php

```diff
@@ -97,20 +97,18 @@
 
                         @if (($setting->slack_channel!='') && ($setting->slack_endpoint))
```

Comment:
This means if your Slack settings are currently blank, you'll *never* get the 'test' button to show up - and the save button is grayed out because it hasn't been tested yet.

I think instead we should *always* show the test button, and we can definitely do something where it's disabled in the initial screen-load until all of the fields are filled. Not exactly sure.

Reply:
I disagree on always showing the test button. If there is incomplete data to send a slack request, we shouldn't attempt it. 

We *could* potentially disable it until all fields are filled in. We'd need an onFocus or onKeyChange to sort that though, so we'd need to pull that logic from server side in the blade to handled only via JS

Reply:
We could also potentially listen to see if there *were* fields when the page loaded and someone deleted them, and add a dynamic checkbox that confirms that they wish to remove their slack integration. On THAT click, if all other fields are empty, then save button becomes enabled again.

Reply:
(We would not have to store that checkbox data or anything, just use it as a UI tool to override the save button being disabled.)